### PR TITLE
support new features

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

File: redisearch/query.py [hunk 1]
```diff
@@ -42,6 +43,15 @@ def limit_ids(self, *ids):
         self._ids = ids
         return self
 
+    def returning(self, *fields):
```
Comment (dvirsky) [inline redisearch/query.py]: rename to `return_fields` please

File: test/test.py [hunk 2]
```diff
@@ -129,7 +132,11 @@ def testClient(self):
                 self.assertEqual(len(subset), docs.total)
                 ids = [x.id for x in docs.docs]
                 self.assertEqual(set(ids), set(subset))
-                
+
+                for doc in client.search(Query('henry king').returning('play', 'nonexist')).docs:
+                    doc.nonexist
```
Comment (dvirsky) [inline test/test.py]: there is no assertion here

File: redisearch/auto_complete.py [hunk 3]
```diff
@@ -4,22 +4,56 @@ class Suggestion(object):
...
         self.score = score
+        self.payload = payload
     
     def __repr__(self):
 
         return self.string
 
+
+class SuggestionParser(object):
+    def __init__(self, with_scores, with_payloads, ret):
```
Comment (dvirsky) [inline redisearch/auto_complete.py]: Add docstring detailing what this does please.

File: test/test.py (hunk 2 above)
Comment (mnunberg) [inline test/test.py]: This is deliberate. I'm unsure what exactly would be included in the field, but I could be sure that trying to access this field would *not* result in an exception (e.g. AttributeError)

Comment (dvirsky) [inline test/test.py]: you do know that doc.nonexist should be None and that doc.play should have content in it

File: redisearch/query.py (hunk 1 above)
Comment (mnunberg) [inline redisearch/query.py]: Done

File: test/test.py (hunk 2 above)
Comment (mnunberg) [inline test/test.py]: Done

File: redisearch/auto_complete.py (hunk 3 above)
Comment (mnunberg) [inline redisearch/auto_complete.py]: Done

**Pull request merged**